src/observable.ts

```diff
@@ -1,22 +1,14 @@
 import { extractPromise, getProxy } from './ObservableObject';
 import { ObservablePrimitiveClass } from './ObservablePrimitive';
 import { isActualPrimitive, isPromise } from './is';
-import type {
-    Observable,
-    ObservableObjectOrArray,
-    ObservablePrimitive,
-    ObservableRoot,
-    PromiseInfo,
-} from './observableInterfaces';
+import type { Observable, ObservablePrimitive, ObservableRoot, PromiseInfo } from './observableInterfaces';
 import { NodeValue } from './observableInterfaces';
 
-function createObservable<T>(value: Promise<T>, makePrimitive?: true): ObservablePrimitive<T & PromiseInfo>;
-function createObservable<T>(value?: T, makePrimitive?: true): ObservablePrimitive<T>;
-function createObservable<T>(
-    value?: Promise<T>,
-    makePrimitive?: boolean,
-): ObservablePrimitive<T & PromiseInfo> | ObservableObjectOrArray<T & PromiseInfo>;
-function createObservable<T>(value?: T, makePrimitive?: boolean): ObservablePrimitive<T> | ObservableObjectOrArray<T> {
+type MaybePromise<T> = NonNullable<T> extends Promise<infer U> ? (U & PromiseInfo) | Extract<T, undefined> : T;
```

Comment:
I am wondering if we should add variant `PromiseObservable` to the `Observable` union type, so that you could just do `Observable<Promise<Something>>` and it would be able to infer the correct type. 

For now I created a utility type